### ISSUE TITLE
Bug: Successful vyper compile doesn't mean a successful boa load

### ISSUE THREAD

**Issue by PatrickAlphaC** [vyperlang/titanoboa#72]
My contract compiles fine, here is the contract:

```python
# SPDX-License-Identifier: MIT
# @version ^0.3.7

# import contract.SimpleStorage as SimpleStorage # This imports it as an interface!
from . import SimpleStorage as SimpleStorage

# interface SimpleStorage:
#     def store(): nonpayable
#     def retrieve() -> uint256: view
#     def add_person(): nonpayable

contract_to_copy: address
list_of_simple_storage_contracts: public(DynArray[SimpleStorage, 10])

@external
def __init__(_contract_to_copy: address):
    self.contract_to_copy = _contract_to_copy


@external
def create_simple_storage_contract():
    deployed_simple_storage: address = create_copy_of(self.contract_to_copy)
    self.list_of_simple_storage_contracts.append(SimpleStorage(deployed_simple_storage))

@external
def storage_factory_store(_simple_storage_index: uint256, _simple_storage_number: uint256):
    # Address 
    # ABI
    self.list_of_simple_storage_contracts[_simple_storage_index].store(_simple_storage_number)

@external
@view
def storage_factory_retrieve(_simple_storage_index: uint256) -> uint256:
    return self.list_of_simple_storage_contracts[_simple_storage_index].retrieve()

```

```bash
 vyper StorageFactory.vy 
0x60206101b66000396000518060a01c6101b157604052346101b15760405160005561017e6100326100003961017e610000f36003361161000c57610166565b60003560e01c3461016c57634ed7246e8118610089576004361061016c57600054803b59811561016c578160381b6a620000003d81600b3d39f317815281600060208301853c600b8201601582016000f0801561016c579050905090506040526001546009811161016c5760018101600155604051816002015550005b636de8b70b81186100e1576044361061016c5760043560015481101561016c5760020154636057361d604052602435606052803b1561016c57600060406024605c6000855af16100de573d600060003e3d6000fd5b50005b63b24a471d8118610137576024361061016c57602060043560015481101561016c5760020154632e64cec1604052602060406004605c845afa610129573d600060003e3d6000fd5b60203d1061016c5760409050f35b63d479dfac8118610164576024361061016c5760043560015481101561016c576002015460405260206040f35b505b60006000fd5b600080fda165767970657283000307000b005b600080fd
```

But, boa doesn't load it fine:
```
$ python
Python 3.10.6 (v3.10.6:9c7b4bd164, Aug  1 2022, 17:13:48) [Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import boa
>>> storage_factory = boa.load("StorageFactory.vy")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/xxxxx/site-packages/boa/interpret.py", line 40, in load
    return loads(f.read(), *args, name=filename, **kwargs)
  File "/xxxxx/site-packages/boa/interpret.py", line 48, in loads
    return d.deploy(*args, **kwargs)
  File "/xxxxxsite-packages/boa/contract.py", line 70, in deploy
    return VyperContract(self.compiler_data, *args, **kwargs)
  File "/xxxxx/site-packages/boa/contract.py", line 368, in __init__
    fns = {fn.name: fn for fn in self.global_ctx._function_defs}
  File "/xxxxx/site-packages/boa/contract.py", line 449, in global_ctx
    return self.compiler_data.global_ctx
  File "/xxxxx/site-packages/vyper/compiler/phases.py", line 117, in global_ctx
    return generate_global_context(self.vyper_module_folded, self.interface_codes)
  File "/xxxxx/site-packages/vyper/compiler/phases.py", line 107, in vyper_module_folded
    module, storage_layout = self._folded_module
  File "/xxxxx/lib/python3.10/functools.py", line 981, in __get__
    val = self.func(instance)
  File "/xxxxx/site-packages/vyper/compiler/phases.py", line 101, in _folded_module
    return generate_folded_ast(
  File "/xxxxx/site-packages/vyper/compiler/phases.py", line 229, in generate_folded_ast
    validate_semantics(vyper_module_folded, interface_codes)
  File "/xxxxx/site-packages/vyper/semantics/validation/__init__.py", line 10, in validate_semantics
    add_module_namespace(vyper_ast, interface_codes)
  File "/xxxxx/site-packages/vyper/semantics/validation/module.py", line 38, in add_module_namespace
    ModuleNodeVisitor(vy_module, interface_codes, namespace)
  File "/xxxxx/site-packages/vyper/semantics/validation/module.py", line 83, in __init__
    err_list.raise_if_not_empty()
  File "/xxxxx/site-packages/vyper/exceptions.py", line 22, in raise_if_not_empty
    raise VyperException("\n\n".join(err_msg))
vyper.exceptions.VyperException: Compilation failed with the following errors:

UnknownType: No builtin or user-defined type named 'SimpleStorage'. 
  contract "StorageFactory.vy:13", line 13:50 
       12 contract_to_copy: address
  ---> 13 list_of_simple_storage_contracts: public(DynArray[SimpleStorage, 10])
  ----------------------------------------------------------^
       14


UndeclaredDefinition: Unknown interface: SimpleStorage. Did you mean 'ERC20Detailed', or maybe 'ERC165'?
  contract "StorageFactory.vy:5", line 5:0 
       4 # import contract.SimpleStorage as SimpleStorage # This imports it as an interface!
  ---> 5 from . import SimpleStorage as SimpleStorage
```

**Comment by PatrickAlphaC** [comment]
Simplestorage.vy

```python
# I'm a comment!

# SPDX-License-Identifier: MIT
# @version ^0.3.7
# If remix isn't updated use version: ^0.2.16

my_favorite_number: uint256

struct Person:
    favorite_number: uint256
    name: String[100]

# Dynamic Array/List
list_of_people: DynArray[Person, 5]

# list_of_people: public(Person[5])
# list_of_people_index: uint256

name_to_favorite_number: HashMap[String[100], uint256]

@external
def store(_favorite_number: uint256):
    self.my_favorite_number = _favorite_number

@external
@view
def retrieve() -> uint256:
    return self.my_favorite_number

@external
def add_person(_name: String[100], _favorite_number: uint256):
    new_person: Person = Person({favorite_number: _favorite_number, name: _name})
    # self.list_of_people[self.list_of_people_index] = new_person
    # self.list_of_people_index += 1
    self.list_of_people.append(new_person)

    self.name_to_favorite_number[_name] = _favorite_number

```

**Comment by charles-cooper** [comment]
should be fixed as of a8c196678c4ba4240a73c94335911cf82b8712c8

**Comment by PatrickAlphaC** [comment]
Just tried from master:

```
Traceback (most recent call last):
  File "xxxdeploy_from_storage_factory.py", line 17, in <module>
    import boa
  File "/xxx/site-packages/boa/__init__.py", line 4, in <module>
    from boa.contract import check_boa_error_matches
  File "/xxx/site-packages/boa/contract.py", line 19, in <module>
    import vyper.semantics.validation as validation
```
